### PR TITLE
Add applet to receive test pattern from CMV12K sensor

### PR DESCRIPTION
**Pull request by tpwrules**
This PR adds an applet which programs the CMV12K sensor to produce its test pattern, then receives and validates it. The received pattern will be used to develop and simulate the logic which writes sensor data to DRAM.

To do:
- [x] remove/justify/improve SocMemory hacks
- [x] validate and dump test pattern
- [x] improve handling of sensor register configuration

### CODE REVIEW

File: naps/cores/peripherals/soc_memory.py [hunk 1]
```diff
@@ -66,7 +66,8 @@ def elaborate(self, platform):
             return m
 
         memorymap = MemoryMap()
-        memorymap.allocate("memory", writable=True, bits=self.depth * memorymap.bus_word_width)
+        assert memorymap.bus_word_width == 32
```
Comment (rroohhh) [inline naps/cores/peripherals/soc_memory.py:71]: `>= 32` should be fine here afaik

**Review by rroohhh — CHANGES_REQUESTED** [review]
The `SocMemory` changes should also be reflected in the `test_smoke` test of `soc_memory_test.py`.

For the hack to map the required amount of pages I have a more proper solution at #22.

File: naps/cores/peripherals/soc_memory.py (hunk 1 above)
Comment (tpwrules) [reply naps/cores/peripherals/soc_memory.py:71]: I think it has to be exactly 32, assuming `memorymap.bus_word_width` is exactly the width of `data` in `handle_read` and `handle_write`.

Comment (rroohhh) [reply naps/cores/peripherals/soc_memory.py:71]: why?

Comment (tpwrules) [reply naps/cores/peripherals/soc_memory.py:71]: because the 32 in those functions would have to be changed to `memorymap.bus_word_width`

**Review by rroohhh — APPROVED** [review]

**Pull request merged**